src/dev-app/dev-app/dev-app-theme.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Injectable} from '@angular/core';
+import {OverlayContainer} from '@angular/cdk/overlay';
+
+const isDarkThemeKey = 'isDarkTheme';
+
+@Injectable({providedIn: 'root'})
+export class DevAppTheme {
+  readonly darkThemeClass = 'demo-unicorn-dark-theme';
+  private _elementToTheme: HTMLElement;
+  private _isDark = false;
+
+  constructor(private _overlayContainer: OverlayContainer) {
+    const isDark = localStorage.getItem(isDarkThemeKey);
```

Comment:
We need to wrap these `localStorage` calls in `try/catch`, because some browsers throw errors if local storage access isn't allowed.